xcdat/bounds.py

```diff
@@ -429,3 +432,227 @@ def _validate_axis_arg(self, axis: CFAxisKey):
+        u_bnd = _month_add(l_bnd, 1, calendar)
+        # store
+        time_bnds.append([l_bnd, u_bnd])
+
+    # create dataarray
+    time_bnds = xr.DataArray(  # type: ignore
+        name=f"{time.name}_bnds",
+        data=time_bnds,
+        coords=dict(time=time),
+        dims=[*time.dims, "bnds"],
+        attrs={"xcdat_bounds": "True"},
+    )
+
+    return time_bnds
+
+
+def get_daily_time_bounds(time, frequency=1):
+    """Sets the time bounds to the start and end of the day
+    for each timestep (this corresponds to 00:00:00 of the
+    timepoint day and 00:00:00 on the subsequent day.
+
+    This function will also set sub-daily bounds if the optional
+    frequency argument is greater than 1. For twice-daily data
+    frequency=2. For 6-hourly, 3-hourly, or hourly data, set
+    frequency to 4, 8, and 24, respectively.
+
+    Parameters
+    ----------
+    time : xr.DataArray
+        The temporal coordinate variable for the axis.
+    frequency : int, optional, default 1
+        Flag to note set the number of timepoints per day.
+
+    Returns
+    -------
+    xr.DataArray
+        The daily or sub-daily time bounds array
+
+    Raises
+    ------
+    ValueError
+        If an incorrect ``frequency`` argument is passed. Should be
+        2, 3, 4, 6, 8, 12, or 24.
+
+    Notes
+    -----
+    This function is intended to reproduce CDAT's setAxisTimeBoundsDaily
+    method [1]_.
+
+    References
+    ----------
+    [1] https://github.com/CDAT/cdutil/blob/master/cdutil/times.py#L1093
+    """
+    # get calendar
+    calendar = time.encoding["calendar"]
+
+    # get cftime class to create new cftime objects
+    cf_obj = get_date_type(calendar)
+
+    # loop over time values and compute bounds
+    time_bnds = []
+    if (frequency > 24) | (np.mod(24, frequency)):
+        raise ValueError(
+            "Incorrect `frequency` argument."
+            " Supported values include 2, 3, "
+            "4, 6, 8, 12, and 24."
+        )
+    for step in time.values:
+        # get year / month
+        y, m, d, h = step.year, step.month, step.day, step.hour
+        for f in range(frequency):
+            if f * (24 // frequency) <= h < (f + 1) * (24 // frequency):
+                l_bnd = cf_obj(y, m, d, f * (24 // frequency))
+                u_bnd = l_bnd + datetime.timedelta(hours=(24 // frequency))
+        # store
+        time_bnds.append([l_bnd, u_bnd])
+
+    # create dataarray
+    time_bnds = xr.DataArray(  # type: ignore
+        name=f"{time.name}_bnds",
+        data=time_bnds,
+        coords=dict(time=time),
+        dims=[*time.dims, "bnds"],
+        attrs={"xcdat_bounds": "True"},
+    )
+
+    return time_bnds
+
+
+def get_time_bounds(time):
+    """Sets the time bounds for a time coordinate axis.
+
+    Parameters
+    ----------
+    time : xr.DataArray
+        The temporal coordinate variable for the axis.
+    frequency : int, optional, default 1
+        Flag to note set the number of timepoints per day.
```

Comment:
```suggestion
```

Suggested edit based on @lee1043's catch (frequency is not an input).

Reply:
I fixed this in my latest commit (https://github.com/xCDAT/xcdat/pull/418/commits/031316b5096f7ae625de40974f34dd08026c589c)